presidio-anonymizer/presidio_anonymizer/anonymizers/hash.py

```diff
@@ -1,19 +1,37 @@
 """Hashes the PII text entity."""
+from hashlib import sha256, sha512, md5
 from presidio_anonymizer.anonymizers import Anonymizer
+from presidio_anonymizer.anonymizers.validators import validate_parameter_in_range
 
 
 class Hash(Anonymizer):
-    # TODO implement + test
-    """Hash given text with sha256 algorithm."""
+    """Hash given text with sha256/sha512/md5 algorithm."""
+
+    HASH_TYPE = "hash_type"
+    SHA256 = "sha256"
+    SHA512 = "sha512"
+    MD5 = "md5"
 
     def anonymize(self, text: str = None, params: dict = None) -> str:
         """
         Hash given value using sha256.
 
         :return: hashed original text
         """
-        return params.get("old_text")
+        hash_type = params.get(self.HASH_TYPE, self.SHA256)
+        hash_switcher = {
+            self.SHA256: lambda s: sha256(s),
+            self.SHA512: lambda s: sha512(s),
+            self.MD5: lambda s: md5(s),
+        }
+        return hash_switcher.get(hash_type)(text.encode()).hexdigest()
 
     def validate(self, params: dict = None) -> None:
-        """TODO: [ADO-2544] docstring."""
+        """Validate the hash type is string and in range of allowed hash types."""
+        validate_parameter_in_range(
+            [self.SHA256, self.SHA512, self.MD5],
+            params.get(self.HASH_TYPE, self.SHA256),
```

Comment:
we're defining the default twice, one might forget to update both if changed.
Either allow self.HASH_TYPE to be empty on validate, check that only if exists then validate, or define a default and use both in anonymize and validate

Reply:
done